adr/0002-keycloak-gitlab.md

```diff
@@ -0,0 +1,165 @@
+# ADR: Enhanced Integration of Keycloak with GitLab for Authentication and User Management
+
+**Date:** 2024-03-05
+
+**Status:** Proposed
+
+## Context
+The initiative to integrate Keycloak with GitLab is designed to streamline authentication mechanisms, enhance security, and improve user experience. This integration aims to refine login processes, bolster security protocols, and ensure compliance with Information Assurance (IA) standards. Utilizing Keycloak as the primary Identity and Access Management (IAM) solution alongside GitLab and leveraging OAuth2 for authentication consolidates user management and authentication, simplifies credential management, and enhances the security landscape.
```

Comment:
we aren't choosing oauth in this case as this implies.

Reply:
👍